src/scriptsys/lib_time.c

```diff
@@ -0,0 +1,68 @@
+#include "lua/lua.h"
+#include "lua/lualib.h"
+#include "lua/lauxlib.h"
+#include "scriptsys/script.h"
+#include "common/console.h"
+#include "common/time.h"
+#include "lib_log.h"
+#include "lib_time.h"
+
+static int func_toSec(lua_State* L)
+{
+	lua_Integer ticks = luaL_checkinteger(L, 1);
+	lua_pushnumber(L, Time_Sec(ticks));
+	return 1;
+}
+
+static int func_toMilli(lua_State* L)
+{
+	lua_Integer ticks = luaL_checkinteger(L, 1);
+	lua_pushnumber(L, Time_Milli(ticks));
+	return 1;
+}
+
+static int func_toMicro(lua_State* L)
+{
+	lua_Integer ticks = luaL_checkinteger(L, 1);
+	lua_pushnumber(L, Time_Micro(ticks));
+	return 1;
+}
+
+void lib_time_init(lua_State* L)
+{
+	LUA_LIB(L,
+		LUA_FN(toSec),
+		LUA_FN(toMilli),
+		LUA_FN(toMicro));
+	LUA_LIB_REG_GLOBAL(L, "Time");
+}
+
+void lib_time_update(lua_State* L)
+{
+	u64 frameCount = Time_FrameCount();
+	u64 appStart = Time_AppStart();
+	u64 frameStart = Time_FrameStart();
+	u64 prevFrame = Time_PrevFrame();
+	u64 now = Time_Now();
+	double delta = Time_Deltaf();
+
+	lua_getglobal(L, "Time");
+		lua_pushinteger(L, frameCount);
+		lua_setfield(L, -2, "frameCount");
```

Comment:
What does -2 mean? Or is that a stack position? Just curious for my own understanding : )

Reply:
Negative goes from the top of the stack (just added). So -1 is the very top, -2 is second from the top, etc.